streams/sensitive-data.js

```diff
@@ -4,12 +4,34 @@ const DEFAULT_SENSITIVE_DATA_FRAGMENTS =
   '(mdp|password|authorization|token|pwd|auth)';
 
 module.exports = class SensitiveDataStream {
-  constructor(fragments) {
+  constructor(fragments, patterns = []) {
     this.fragments = fragments || DEFAULT_SENSITIVE_DATA_FRAGMENTS;
-    this.pattern = new RegExp(`"${this.fragments}":"([^"]*)"`, 'ig');
+    this.replacer = '__SENSITIVE_DATA__';
+
+    // If a pattern is provided
+    if (patterns.length) {
+      this.patterns = patterns;
+    } else {
+      // If no pattern provided, then add 2 default regexes
+      this.patterns = [
+        {
+          regex: new RegExp(`"${this.fragments}":"([^"]*)"`, 'ig'), // @Match "mdp":"My super password"
+          substitute: `"$1":"${this.replacer}"`,
+        },
+        {
```

Comment:
I would remove this case from the default regexp to keep the same behaviour as before.